homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -64,200 +57,141 @@
 class WeatherUpdateCoordinator(DataUpdateCoordinator):  # pylint: disable=hass-enforce-coordinator-module
     """Weather data update coordinator."""
 
-    def __init__(self, owm, latitude, longitude, forecast_mode, hass):
+    def __init__(
+        self,
+        owm_client: OWMClient,
+        latitude,
+        longitude,
+        hass: HomeAssistant,
+    ) -> None:
         """Initialize coordinator."""
-        self._owm_client = owm
+        self._owm_client = owm_client
         self._latitude = latitude
         self._longitude = longitude
-        self.forecast_mode = forecast_mode
-        self._forecast_limit = None
-        if forecast_mode == FORECAST_MODE_DAILY:
-            self._forecast_limit = 15
 
         super().__init__(
             hass, _LOGGER, name=DOMAIN, update_interval=WEATHER_UPDATE_INTERVAL
         )
 
     async def _async_update_data(self):
         """Update the data."""
-        data = {}
         async with asyncio.timeout(20):
```

Comment:
Please pass the timeout value to the library and than the aiohttp client request so we don't have two timeouts running

Reply:
Added as optional parameter https://github.com/freekode/pyopenweathermap/blob/main/src/pyopenweathermap/owm_client.py#L14
Removed timeout in coordinator